File: 이건희/level2/배달.py
def solution(N, road, K):
    result = 0
    graph = [[10 ** 9] * (N+1) for _ in range(N+1)]

    for i in range(N + 1):
        graph[i][i] = 0
        
    for i in road:
        graph[i[0]][i[1]] = min(graph[i[0]][i[1]], i[2])
        graph[i[1]][i[0]] = min(graph[i[1]][i[0]], i[2])

    for i in range(1, N + 1):
        for j in range(1, N + 1):
            for k in range(1, N + 1):
                graph[j][k] = min(graph[j][i] + graph[k][i], graph[j][k])

    return len([d for d in graph[1] if d <= K])

File: 이훈일/Programmers/Level2/배달.java
import java.util.*;

class Solution {
    public int solution(int N, int[][] road, int K) {
        int INF = 500001;
        
        int[][] graph = new int[N+1][N+1];
        for(int i = 1; i <= N; i++) {
            for(int j = 1; j <= N; j++) {
                if(i != j) {
                    graph[i][j] = INF;
                }
            }
        }
        
        for(int i = 0; i < road.length; i++) {
            int town1 = road[i][0];
            int town2 = road[i][1];
            int time = road[i][2];
            
            graph[town1][town2] = Math.min(time, graph[town1][town2]);
            graph[town2][town1] = Math.min(time, graph[town2][town1]);
        }
        
        for(int k = 1; k <= N; k++) {
            for(int i = 1; i <= N; i++) {
                for(int j = 1; j <= N; j++) {
                    if(graph[i][j] > graph[i][k] + graph[k][j]) {
                        graph[i][j] = graph[i][k] + graph[k][j];
                    }
                }
            }
        }
        
        int count = 0;
        for(int j = 1; j <= N; j++) {
            if(graph[1][j] <= K) {
                count++;
            }
        }
        
        return count;
    }
}
